### PR TITLE
Add VPC attachment resource

### PR DESCRIPTION
**Pull request by annakhm**
This is the new way of attaching connectivity profile to vpc, hence the profile attribute is also removed from vpc resource

### CODE REVIEW

**Comment by annakhm** [comment]
--- PASS: TestAccResourceNsxtVpc_basic (72.41s)
--- PASS: TestAccResourceNsxtVpc_importBasic (28.18s)

**Review by ksamoray — APPROVED** [review]
LGTM

File: nsxt/provider.go [hunk 1]
```diff
@@ -528,6 +528,7 @@ func Provider() *schema.Provider {
 			"nsxt_vpc_static_route":                     resourceNsxtVpcStaticRoutes(),
 			"nsxt_policy_project_ip_address_allocation": resourceNsxtPolicyProjectIpAddressAllocation(),
 			"nsxt_vpc_dhcp_v4_static_binding":           resourceNsxtVpcSubnetDhcpV4StaticBindingConfig(),
+			"nsxt_vpc_attachment":                       resourceNsxtVpcAttachment(),
 		},
```
Comment (tenthirtyam) [inline nsxt/provider.go:533]: I'd suggest this be the first in the lists of `nsxt_vpc_*` items (as well as ensure all listed resources and data sources are listed alphabetically.)

File: nsxt/resource_nsxt_vpc_attachment.go [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+	return resourceNsxtVpcAttachmentRead(d, m)
+}
+
+func resourceNsxtVpcAttachmentRead(d *schema.ResourceData, m interface{}) error {
+	connector := getPolicyConnector(m)
+
+	id := d.Id()
+	if id == "" {
+		return fmt.Errorf("Error obtaining VpcAttachment ID")
```
Comment (tenthirtyam) [inline nsxt/resource_nsxt_vpc_attachment.go:119]:
```suggestion
		return fmt.Errorf("error obtaining VpcAttachment ID")
```

File: nsxt/resource_nsxt_vpc_attachment.go [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+	return metadata.StructToSchema(elem, d, vpcAttachmentSchema, "", nil)
+}
+
+func resourceNsxtVpcAttachmentUpdate(d *schema.ResourceData, m interface{}) error {
+
+	connector := getPolicyConnector(m)
+
+	id := d.Id()
+	if id == "" {
+		return fmt.Errorf("Error obtaining VpcAttachment ID")
```
Comment (tenthirtyam) [inline nsxt/resource_nsxt_vpc_attachment.go:150]:
```suggestion
		return fmt.Errorf("error obtaining VpcAttachment ID")
```

File: nsxt/resource_nsxt_vpc_attachment.go [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+		return handleUpdateError("VpcAttachment", id, err)
+	}
+
+	return resourceNsxtVpcAttachmentRead(d, m)
+}
+
+func resourceNsxtVpcAttachmentDelete(d *schema.ResourceData, m interface{}) error {
+	id := d.Id()
+	if id == "" {
+		return fmt.Errorf("Error obtaining VpcAttachment ID")
```
Comment (tenthirtyam) [inline nsxt/resource_nsxt_vpc_attachment.go:187]:
```suggestion
		return fmt.Errorf("error obtaining VpcAttachment ID")
```

**Review by tenthirtyam — CHANGES_REQUESTED** [review]

File: nsxt/resource_nsxt_vpc_service_profile.go [hunk 5]
```diff
@@ -24,14 +24,6 @@ var vpcServiceProfileModeValues = []string{
 	model.VpcProfileDhcpConfig_MODE_DEACTIVATED,
 }
 
-var vpcServiceProfileLogLevelValues = []string{
```
Comment (salv-orlando) [inline nsxt/resource_nsxt_vpc_service_profile.go:27]: Was this really unused?

File: nsxt/policy_utils.go [hunk 6]
```diff
@@ -320,7 +320,7 @@ func parseStandardPolicyPath(path string) ([]string, error) {
 			infraPath = false
 		}
 	}
-	if len(segments) <= idx {
+	if len(segments) < idx {
```
[the remark this answers is not in the document]
Comment (annakhm) [reply nsxt/policy_utils.go:323]: Good point, added

File: nsxt/resource_nsxt_vpc_service_profile.go (hunk 5 above)
Comment (annakhm) [reply nsxt/resource_nsxt_vpc_service_profile.go:27]: Yes, these values were also removed from the SDK

**Review by salv-orlando — APPROVED** [review]
LGTM!

**Pull request merged**